packages/stories-block/src/block/components/authorSelection.js

```diff
@@ -41,38 +41,48 @@ import Autocomplete from './autocomplete';
  * @property {()=>void} setAttributes Callable function for saving attribute values.
  */
 
+const defaultQueryProps = {
+  per_page: 100,
+};
+
 /**
  * AuthorSelection component. Used for selecting authors of stories.
  *
  * @param {AuthorSelection} props Component props.
  * @return {*} JSX markup.
  */
 const AuthorSelection = ({ authors: authorIds, setAttributes }) => {
-  const [isInitialized, setIsInitialized] = useState(false);
-  const [authorsList, setAuthorsList] = useState([]);
-  const [authorSuggestions, setAuthorSuggestions] = useState([]);
-
-  useEffect(() => {
-    if (isInitialized || !authorIds?.length) {
-      return;
-    }
-
-    apiFetch({
-      path: addQueryArgs('/wp/v2/users', {
-        per_page: 100,
+  const [authorKeyword, setAuthorKeyword] = useState('');
+
+  const { authorsList } = useSelect(
+    (select) => {
+      const { getUsers } = select(coreStore);
+
+      const query = {
+        ...defaultQueryProps,
         include: authorIds.join(','),
-      }),
-    })
-      .then((users) => {
-        if ('undefined' !== typeof users && Array.isArray(users)) {
-          setAuthorsList(users.map(({ id, name }) => ({ id, value: name })));
-        }
-      })
-      .catch(() => {
-        setAuthorsList([]);
-      })
-      .finally(() => setIsInitialized(true));
-  }, [isInitialized, authorIds]);
+      };
+      return {
+        authorsList: getUsers(query) || [],
+      };
+    },
+    [authorIds]
+  );
+
+  const { authorSuggestions } = useSelect(
```

Comment:
IIRC we discussed merging the two `useSelect()` calls into one, for consistency

Reply:
Sorry, I didn't put a comment. Merging these means more queries. As like hooks, it means then depencies changes, then it is rerun. We don't want there here, as it means more database queries.